{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###\n",
    "\n",
    "By product of the SNP pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[NbConvertApp] Converting notebook MergeViralAndBacterial.ipynb to notebook\n",
      "[NbConvertApp] Executing notebook with kernel: python3\n",
      "[NbConvertApp] ERROR | Error while converting 'MergeViralAndBacterial.ipynb'\n",
      "Traceback (most recent call last):\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/nbconvertapp.py\", line 393, in export_single_notebook\n",
      "    output, resources = self.exporter.from_filename(notebook_filename, resources=resources)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/exporters/exporter.py\", line 174, in from_filename\n",
      "    return self.from_file(f, resources=resources, **kw)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/exporters/exporter.py\", line 192, in from_file\n",
      "    return self.from_notebook_node(nbformat.read(file_stream, as_version=4), resources=resources, **kw)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/exporters/notebook.py\", line 31, in from_notebook_node\n",
      "    nb_copy, resources = super(NotebookExporter, self).from_notebook_node(nb, resources, **kw)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/exporters/exporter.py\", line 134, in from_notebook_node\n",
      "    nb_copy, resources = self._preprocess(nb_copy, resources)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/exporters/exporter.py\", line 311, in _preprocess\n",
      "    nbc, resc = preprocessor(nbc, resc)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/preprocessors/base.py\", line 47, in __call__\n",
      "    return self.preprocess(nb, resources)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/preprocessors/execute.py\", line 262, in preprocess\n",
      "    nb, resources = super(ExecutePreprocessor, self).preprocess(nb, resources)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/preprocessors/base.py\", line 69, in preprocess\n",
      "    nb.cells[index], resources = self.preprocess_cell(cell, resources, index)\n",
      "  File \"/cellar/users/btsui/anaconda3/lib/python3.6/site-packages/nbconvert/preprocessors/execute.py\", line 286, in preprocess_cell\n",
      "    raise CellExecutionError.from_cell_and_msg(cell, out)\n",
      "nbconvert.preprocessors.execute.CellExecutionError: An error occurred while executing the following cell:\n",
      "------------------\n",
      "import pandas as pd\n",
      "import numpy as np\n",
      "import re\n",
      "import os\n",
      "import math\n",
      "from multiprocessing import Pool\n",
      "\n",
      "## init\n",
      "mySpecie='Homo_sapiens'\n",
      "\n",
      "##change this dir to point to the updated csv\n",
      "full_meta_dir=\"/cellar/users/btsui/Project/METAMAP/notebook/Parsing/sra_dump.csv\"\n",
      "inSrrDir='/nrnb/users/btsui/Data/all_seq/snp/'\n",
      "tmp_dir='/nrnb/users/btsui/Data/all_seq/tmp/'\n",
      "\n",
      "inAllFames=pd.Series(os.listdir(inSrrDir))\n",
      "\n",
      "#os.system('rm '+tmp_dir+'*')\n",
      "\n",
      "perFaStatS=inAllFames[inAllFames.str.contains('per_fa_record_stat.txt.gz')].values\n",
      "\n",
      "TEST=False\n",
      "if TEST:\n",
      "    toRunSrrs=perFaStatS[:10]\n",
      "    chunkSize=5\n",
      "    nThread=1\n",
      "else:\n",
      "    toRunSrrs=perFaStatS\n",
      "    chunkSize=1000\n",
      "    nThread=64\n",
      "\n",
      "## get the microbes FA\n",
      "tmpDf=pd.read_csv(inSrrDir+'ERR1497972_per_fa_record_stat.txt.gz',sep='\\t',header=None)\n",
      "ignoreList=['1', '2', '3', '4', '5', '6', '7',\n",
      "       '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18',\n",
      "       '19', '20', '21', '22', 'X', 'Y', 'MT', '*']\n",
      "\n",
      "microbes_fa=tmpDf[0][~tmpDf[0].isin(ignoreList)].values\n",
      "\n",
      "\n",
      "#inFname=toRunSrrs[0]\n",
      "def parseFile(inFname):\n",
      "    \"\"\"\n",
      "    input: inAllFames\n",
      "    for each file, merge the microbe data file\n",
      "    \"\"\"\n",
      "    #The output is TAB-delimited with each line consisting of reference sequence name, sequence length, # mapped reads and # unmapped reads. \n",
      "    fDir=inSrrDir+inFname\n",
      "    tmpDf2=pd.read_csv(fDir,sep='\\t',header=None)\n",
      "    tmpDf2.columns=['fa_name','','n_mapped_reads','']\n",
      "    tmpS=tmpDf2[~tmpDf2['fa_name'].isin(ignoreList)].set_index('fa_name')['n_mapped_reads']\n",
      "    tmpS.name=inFname\n",
      "    #tmpS.index.name=inFname\n",
      "    return tmpS\n",
      "\n",
      "\n",
      "\n",
      "#tmpS=parseFile(inFname)\n",
      "\n",
      "\n",
      "def mergeSrrsL(i):\n",
      "    tmpL=[]\n",
      "    failedSrrsL=[]\n",
      "    for srr in toRunSrrs[i:(i+chunkSize)]:\n",
      "        try:\n",
      "            tmpL.append(parseFile(srr))\n",
      "        except :\n",
      "            print 'failed: '+srr\n",
      "            failedSrrsL.append(srr)\n",
      "    tmpMergedDf=pd.DataFrame(tmpL).T\n",
      "    #print tmpMergedDf.shape\n",
      "    #tmpMergedDf=pd.concat([parseSrr(srr) for srr in toRunSrrs[i:(i+chunkSize)]])\n",
      "    reorderedDf=tmpMergedDf.sort_index()\n",
      "    if TEST:\n",
      "        print tmp_dir+str(i)+'.pickle.gz'\n",
      "    reorderedDf.to_pickle(tmp_dir+str(i)+'.pickle.gz',compression='gzip')\n",
      "    return failedSrrsL\n",
      "\n",
      "Chunks=np.arange(0, len(toRunSrrs),chunkSize)\n",
      "if TEST:\n",
      "    failed_srr_l=map(mergeSrrsL,Chunks.tolist())\n",
      "else:\n",
      "    from multiprocessing import Pool\n",
      "    p=Pool(nThread)\n",
      "    ### sweep for uncompleted chunks\n",
      "    failed_srr_l=p.map(mergeSrrsL,Chunks.tolist())\n",
      "    p.close()\n",
      "\n",
      "\n",
      "#testDf3=pd.read_pickle('/nrnb/users/btsui/Data/all_seq/tmp/0.pickle.gz')\n",
      "\n",
      "myL=[]\n",
      "for fname in os.listdir(tmp_dir):\n",
      "    tmpDf10=pd.read_pickle(tmp_dir+fname).astype(np.float32)\n",
      "    myL.append(tmpDf10)\n",
      "mergedDf=pd.concat(myL,axis=1)\n",
      "\n",
      "mergedDf.columns=mergedDf.columns.str.replace('_per_fa_record_stat\\.txt\\.gz','')\n",
      "------------------\n",
      "\n",
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-4-9b549970319c>\"\u001b[0;36m, line \u001b[0;32m68\u001b[0m\n",
      "\u001b[0;31m    print 'failed: '+srr\u001b[0m\n",
      "\u001b[0m                   ^\u001b[0m\n",
      "\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m Missing parentheses in call to 'print'. Did you mean print('failed: '+srr)?\n",
      "\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print('failed: '+srr)? (<ipython-input-4-9b549970319c>, line 68)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "%%bash \n",
    "jupyter nbconvert --execute --to notebook --ExecutePreprocessor.timeout=-1 --inplace MergeViralAndBacterial.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!mv ./../XGS_WGS/MergeViralAndBacterial.ipynb ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#XGS_WGS/MergeViralAndBacterial.ipynb"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
